backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/log/ModularContractLogDAO.java

```diff
@@ -181,7 +213,7 @@ public ModularContractLogEntryId reverse(@NonNull final LogEntryReverseRequest r
 			reversedLog.setQty(reversedLog.getQty().negate());
 		}
 
-		if (reversedLog.getAmount() != null)
+		if (reversedLog.getAmount() != null && !BigDecimal.ZERO.equals(reversedLog.getAmount()))
```

Comment:
instead of comparing with ZERO pls use signum() != 0

Reply:
also as a side note about reverse() method.... DAO is not a place for such a method, but service layer.
More then method shall get the `ModularContractLogEntry` 
then call some ModularContractLogEntry.createReversal() which returns another ModularContractLogEntry
and then ask the DAO to save it.